salt/client/ssh/__init__.py

```diff
@@ -614,15 +657,18 @@ def handle_ssh(self, mine=False):
                     self.targets[host],
                     mine,
                 )
-                try:
-                    logging._acquireLock()
-                    routine = Process(target=self.handle_routine, args=args)
-                    routine.start()
-                except:
-                    pass
-                finally:
-                    logging._releaseLock()
+                routine = Process(target=self.handle_routine, args=args)
+                # Explicitly call garbage collector to prevent possible segfault
+                # in salt-api child process. (bsc#1188607)
+                gc.collect()
+                routine.start()
                 running[host] = {"thread": routine}
+                with salt.utils.files.flopen(self.session_flock_file, "w"):
```

Comment:
Same question here, isn't this breaking in case lock file is there already?

Reply:
`salt.utils.files.flopen` used as an atomic operation to properly put the correct data to the cache file with preventing race conditions